# Introduction

## The G-Quadruplex

### What is a Quadruplex?

The genome is often referred to as the "blueprint" for life. This metaphor suggests a static set of instructions, which simply encodes data and does not change its form. In reality, however, chromosomes are highly dynamic structures which are able to undergo various covalent modifications to the DNA and proteins, as well as change topologies on a global and local scale. At the global level, changes in how the chromatin is packed result in the closing off or opening up of specific regions, changing the level of transcription of the genes contained within them through various means . Genes which are far apart in sequence space can be brought together through looping to allow co-regulation. At the smaller scale, the DNA itself is able to fold into a number of different shapes, including various types of duplex, triplex, and quadruplex. Examples include B-DNA (the classic double helix), A-DNA (duplex), R-loops (triplex) [@Roberts1992, @Korzheva2000], i-motifs (quadruplex) [@Gehring1993, @Tua2000, @Zeraati2018] and G-Quadruplexes [@Gellert1962, @Sen1988, @Sundquist1989, @Hansel2016]. These structures have different relative stabilities depending on the conditions of their local environment, e.g. the local concentration of solutes, complementary RNAs, or stabilising proteins, the level of molecular crowding, or the intracellular pH. Furthermore, some structures form only in DNA containing specific sequence attributes, such as high GC content. Whilst non-B forms of DNA have been known to form *in vitro* for some time, it is only recently that evidence of their formation *in vivo*, and their effects on biological systems, has begun to accumulate.

One of the more well studied non-B DNA/RNA forms is the G-Quadruplex (G4), a Guanine rich, four stranded DNA helix. The properties of guanine which allow G4s to form were first hinted at by German chemist Ivar Bang in 1910, more than four decades before Watson, Crick and Franklin deduced the structure of the double helix [@Bang1910]. Bang noted that guanosine nucleotides in concentrations of around 25 mg/ml will form a viscous gel. It was not until 1962, however, that Gellert et al. were able to use the technique of X-ray diffraction to identify the interactions which caused this property. They noted that guanine monomers were able to interact to form square, planar quartets, which then stacked to form a helical structure [@Gellert1962].

Many more recent studies have shown conclusively that far from being just an unusual property of monomeric guanine, square planar "G-quartets" can form from polymeric DNA, so long as it contains a high enough local concentration of guanines. The guanines in a quartet interact through non-canonical Hoogsteen base pairing, with two hydrogen bonds occurring between each adjacent base (i.e. each base participates in four hydrogen bonds, for a total of eight bonds per quartet) (Fig \ref{g4_struct}a) [@Gellert1962, @Sen1988, @Sundquist1989]. These quartets can then stack on top of one another through hydrophobic interactions (Fig \ref{g4_struct}b). The number of stacked quartets in a G4 is usually referred to as the number of tetrads. The stability of G4s is generally proportional to the number of tetrads they contains. Mono or divalent cations, usually potassium, fit into the central channel of the G4. These sit equidistant from each guanine of each quartet, and between each pair of adjacent tetrads. The number of potassium cations is therefore one less than the number of tetrads (Fig \ref{g4_struct}b). Potassium is strongly stabilising of G4s, and so the stability of G4s is dependent upon potassium concentration [@Schultze1999].

The structure of G4s is highly polymorphic. Quadruplexes can form between multiple molecules (intermolecular) or from the same molecule (intramolecular). Intermolecular G4s tend to be less common *in vivo* than *in vitro* however as the effective concentration of G4 forming motifs *in vivo* is generally much lower than under experimental *in vitro* conditions [@Huppert2005]. On top of this, G4s can fold with DNA strands in the same orientation (parallel), in different orientations (anti-parallel) or in a mixed hybrid conformation (Fig \ref{g4_struct} c) [@Parkinson2002, @Phan2003, @Ambrus2006]. Which conformation is chosen can be dependent upon the sequence from which the G4 is formed, as well as environmental conditions. The loops which connect the G-rich "pillars"   of the G4 can be connected laterally (resulting in anti-parallel conformation), diagonally (also anti-parallel) or through a "propeller" like fold (resulting in parallel strands). An anti-parallel G4 with only lateral loops is generally referred to as a "chair" like G4, whilst an anti-parallel G4 with a diagonal loop is referred to as "basket" like (Fig \ref{g4_struct} c). Many G4 forming sequences will fold into multiple conformations with different rates, resulting in two or more subpopulations of folded G4 from molecules of the same sequence. The human telomeric sequence has been found to adopt both parallel, anti-parallel and hybrid forms *in vitro*, suggesting a dynamic population [@Parkinson2002, @Phan2003, @Ambrus2006].

\newpage

![**Structure of a G-Quadruplex** **a)** The molecular structure of a G-quartet. Four Guanosines (only guanine base is shown, sugar-phosphate is represented as R) interact through Hoogsteen base pairing around a central monovalent cation. **b)** Cartoon showing the basic structure of a three tetrad G4. Three G-quartets are stacked through interactions between delocalised pi electrons. The structure is made up of four pillars of homopolymeric G-runs (shown in orange) joined by loop sequences. Potassium cations (shown in blue) sit between each tetrad. **c)** Cartoons showing how loop arrangement can contribute to the structural polymorphism of G4s. Loops can be lateral, diagonal or propeller like, resulting in anti-parallel, anti-parallel, and parallel G4s respectively. Anti-parallel G4s with all lateral loops are referred to as "chair"-like, whilst anti-parallel G4s with a diagonal loop are referred to as basket like. G4s can also contain a mix of parallel and anti-parallel strands. \label{g4_struct}](figures/g4_structure.svg)

\newpage

![**The Human Telomeric Sequence forms a G-Quadruplex** **a)** top and **b)** side view of a parallel G4 structure solved by X-ray crystallography (PDB: 1KF1, Parkinson et al. 2002). The sequence used is (TTAGGG)4, corresponding to four of the human telomeric repeat. Gs are coloured in yellow, As in green and Ts in orange. Potassium cations are coloured in purple.](figures/xray_structure.svg){height=750px}

\newpage

### G-Quadruplex Prediction from Sequence

Genomic G4s form in sequences with high G4 content and high GC skew. Because of the dependence of G4 structure on sequence, it is theoretically possible to predict the genome-wide prevalence of G4s from sequence information alone, assuming that other conditions such as potassium concentration are held constant. The first attempt to characterise putative G4s (PG4s) at whole genome scale was conducted by Huppert and Balasubramanian in 2005. They formulated rules describing the general patterns that PG4 forming sequences tend to follow. Their first observation was that intermolecular G4s are unlikely to be common *in vivo* due to low strand concentration of the DNA. They also noted that the pillars of the G4 tended to be formed from contiguous guanine homopolymers, or G-runs. There have to be four such G-runs in close proximity to create a PG4, and the length of the shortest G-run will determine the maximum number of stacked tetrads which can be formed. A minimum number of 3 tetrads was suggested for prediction: whilst 2 tetrad G4s are possible, they are less stable. Finally, they suggested that to make folding of the G4 favourable, the length of the loop sequences connecting the G-runs should be relatively short. They suggested, using evidence from molecular modelling and CD spectroscopy, an upper limit of 7bp. Again, whilst loops of much longer length are possible, they were thought likely to be unstable. Their observations were combined to create the folding pattern $G_XN_{1-7}G_XN_{1-7}G_XN_{1-7}G_X$, where $X \geq 3$. This was named the Quadparser method (Huppert and Balasubramanian 2005), and can be applied to search genomes using simple regular expression machinery.

The Quadparser method has been successfully used to identify G4s in many organisms, however the it is not perfect and results in many false negatives as well as false positives. Various adjustments can and have be made to the pattern, including increasing loop length to a maximum of 12bp, allowing two tetrad PG4s, and allowing short bulges in G-runs. These tend to increase the recall of the method but also greatly increase the number of false positives. Other methods have been proposed, such as G4Hunter (Bedrat et al. 2016), which allow PG4s to be given a numeric score based on the GC content and skew of the sequence. G4Hunter is generally performed using a sliding window between 20 and 40bp in length, and is evaluated for each window by the following method:

```python
score = 0
for base, run_length in run_length_encode(sequence):
    if base == 'G':
        score += min(run_length, 4) ** 2
    elif base == 'C':
        score -= min(run_length, 4) ** 2
    else:
        pass
score /= len(sequence)
```

Sequences which have high PG4 forming ability on the positive strand will therefore be given strong positive scores, whilst sequences with PG4 forming ability on the negative strand will be given strong negative scores. A threshold value is chosen below which to filter out non-PG4 forming sequences. Bedrat et al argued that this method was an improvement over the Quadparser technique because it was more flexible, however it also results in a much greater number of false positives when applied to a whole genome, since there are no constraints on how the G-runs are arranged in the windowed sequence.

\newpage

### Methods used in the characterisation of G-Quadruplexes

Gellert et al. first characterised G4 structure using X-ray diffraction of fibres from dehydrated guanine gels (Gellert et al. 1962). Since then, biophysical techniques have become key in the study of G4 structures *in vitro*. Since the advent of chemical DNA oligonucleotide synthesis in the 1980s, it has become relatively cheap to order high purity single stranded oligonucleotides for PG4 sequences, and produce micromolar concentration solutions which can be probed by CD spectroscopy or NMR.

Circular dichroism (CD) spectroscopy utilises the difference in absorbance of circularly polarised light by molecules with chiral structures (i.e. with non-superimposable mirror images). Parallel and anti-parallel G4s both exhibit unique CD absorbance spectra which are distinct from the spectra of disordered single stranded DNA. Solutions containing multiple subpopulations of different parallel and antiparallel G4s will produce spectra which are more complex to interpret, but are still clearly distinct from unordered DNA. Melting temperatures of G4s can be determined using CD or UV spectroscopy temperature gradients measured at 295nm.

Nuclear magnetic resonance spectroscopy (NMR), specifically Proton-exchange spectroscopy (1H-NMR), can also be used to identify G4 DNA. NMR is conducted in deuterated water, since deuterium has a spin of 1 and therefore does not contributed to the NMR spectrum. In single or double stranded DNA, imino protons in the guanine nucleotides will be exchanged with the solvent on short timescales, resulting in loss of the imino proton signal as they are replaced with deuterated protons. In G4 DNA, on the other hand, imino protons are located centrally within the G4 structure, and are therefore protected from exchange. This means that the 1H spectra can be used to distinguish . This does not however identify whether the G4 has parallel or anti-parallel topology. Further characterisation can be conducted using Nuclear Overhauser effect spectroscopy (NOESY) to identify spatial relationships between protons in the G4.

Finally, since folded G4s with short loops and flanking sequence are relatively globular, a number of G4 structures have been crystallised from oligomers. The structures of these crystals have then been solved by X-ray crystallography.

Studies of the structure of G4s has led to the development of a variety of G4-binding ligands. These have a wide range of structures and bind to the various G4 topologies with different strengths. The major classes of G4 binding ligands are: external stacking ligands, which have delocalised pi electron systems, and stack on the the hydrophobic surfaces of the outer G4 tetrads; intercalating ligands, which bind in the groove between stacked tetrads; and external groove binding ligands, which insert into the groove of the G4 helix, or interact with loop sequences (Chen et al 2014). Small molecules which bind G4s include: porphyrins, such as N-methyl-mesoporphyrin (NMM), an external stacking ligand; Pyridostatin, another external stacking molecule specifically designed for G4 binding; and Berberine, a naturally occurring alkaloid and G4 external stacking agent (Fig \ref{drugs}). These molecules have been used to study G4s *in vitro* and also *in vivo*.

\newpage

![**G-Quadruplex Stabilising Ligands** Structures and mode of action of Pyridostatin, NMM, and Berberine. Pyridostatin is thought to bind to all G4s equally well (Müller et al 2010), whilst NMM is known to prefer parallel G4s (Nicoludis et al. 2012). Berberine has been reported to bind to both parallel and anti-parallel G4s (Bazzicalupi et al. 2013, Li et al 2017). Crystal structures of human telomeric DNA (parallel G4 form) in complex with NMM and Berberine are from PDB entries 4FXM and 3R6R respectively (Nicoludis et al. 2012, Bazzicalupi et al. 2013). Potassium ions and solvent molecules have been hidden for visualisation purposes. \label{drugs}](figures/drugs.svg)

\newpage

Whilst biophysical methods have led to a wealth of data on G4 formation *in vitro*, biological evidence of G4 formation has been a much later development. One common approach to studying the effect of G4 stabilisation on biological processes is to treat cells or organisms with G4 binding ligands such as NMM or pyridostatin. This has been shown to have various effects on replication, genome stability, transcription and development. Naturally fluorescent or fluorophore labelled small molecules are commonly used to localise G4s by microscopy, or study their *in vitro* folding by single molecule Förster resonance energy transfer (smFRET) (Maleki et al. 2017). Biotinylated pyridostatin has also been used to pull down G4 DNA structures from human DNA (Müller et al. 2010).

Moving beyond small molecules, Biffi et al developed an antibody which specifically recognises G4 DNA using synthetic phage display technology (Biffi et al. 2013). They used the antibody, named BG4, in immunofluorescence experiments to visualise G4s in human chromatin (Biffi et al 2013). More recently, it was used in Chromatin Immunoprecipitation sequencing (ChIP-seq) experiments to identify the specific regions of human chromatin where folded G4s occur (Hänsel-Hertsch et al. 2016).

A number of techniques have been employed for whole genome or transcriptome mapping of G4s. In a method they named G4-seq, Chambers et al. introduced potassium or pyridostatin into the buffer of Illumina sequencing-by-synthesis reactions (Chambers et al. 2015). This resulted in G4 formation in the single stranded DNA fragments, which caused stalling of DNA polymerase, resulting in sequencing errors. They conducted this method on single stranded DNA derived from the human genome. Clusters of identical reads generated by bridge amplification in an Illumina flow cell were first sequenced normally. The products of the first sequencing-by-synthesis reaction were then washed off, and the clusters were resequenced in the presence of the G4 binding agents. This yielded an initial high quality, mappable read, and an error prone read. When they mapped the initial reads to the genome, the number of sequencing errors in each position was considered an indicator of the G4 forming potential. The authors showed using this technique that only 30% of experimentally observed G4s in the human genome conform to the Quadparser motif

Yoshida et al. also used a similar method to identify G4 clusters in human genomic DNA, by PCR amplifying sequences in the presence and absence of the G4 binding ligand telomestatin (Yoshida et al. 2018). They showed that regions which contained G4s were amplified with lower efficiency in the presence of telomestatin, due to polymerase stalling events. This resulted in a quantifiable reduction in the number of reads mapping to G4 containing regions of the genome in the drug positive samples, relative to the drug negative samples. The authors identified that regions of the genome which were predicted to form G4 clusters, rather than just single G4s, were most effective at stalling polymerase.

G4s which form in RNA can also be mapped globally, using a technique called rG4-seq, developed by Kwok et al (Kwok et al 2016). This method again utilised stalling at G4s, in the presence of potassium or pyridostatin, this time by the RNA templated DNA polymerase Reverse transcriptase (RT). They identified positions in mRNAs where a reproducible drop in reads occurred in samples where RT mediated DNA synthesis was conducted in the G4 stabilising conditions, relative to unstabilised controls.

### G-Quadruplex stability prediction using Machine Learning:

A growing number of G4 forming oligonucleotide sequences have now been characterised *in vitro* by various methods, particularly by CD spectroscopy and UV melting, and the melting temperatures have been published online. The number of these is now great enough that several groups have utilised them to train machine learning models to predict G4 forming potential. Stegle et al. used a Gaussian Process model incorporating extracted features from Quadparser conforming G4s (Stegle et al. 2009). These features were the number of tetrads, the length of each of the three loops, the total loop length, and the frequencies of adenine, cytosine and guanine in the sequence, as well as the raw sequence itself. A second kernel which incorporated features about the conditions the melting temperature was acquired under, i.e. the concentrations of potassium, sodium, ammonium, and magnesium ions, was also used in the model. They trained this model on a set of 260 DNA G4 melting temperatures which were acquired from a literature search. In a cross validation experiment using 100 random 50% hold out splits, the authors were able to achieve a good level of test set accuracy with an average of 80% of predictions within 5 degrees of the true melting temperature (Stegle et al. 2009). Furthermore, their model was interpretable, and they were able to identify tetrad number and the length of the central loop as the most important sequence features in PG4 stability. The authors employed active learning to identify candidates from human promoter sequences with high uncertainty in the model, and used CD spectroscopy to characterise them.

Whilst Stegle et al.'s model was successful on Quadparser conforming motifs, it is estimated that ~70% of G4s in the human genome do not conform to this motif (Chambers et al. 2015). More recently, Garant et al. published a method for predicting RNA G4s which was trained on a set of 368 experimentally determined sequences, 149 of which were G4 positive and 179 of which were G4 negative (Garant et al. 2017). From these sequences the trinucleotide contents were extracted, and used to train a densely connected multi-layer perceptron model, with a single hidden layer containing 35 nodes. This trinucleotide trained model had the advantage being more flexible for G4s that do not conform to the Quadparser motif. Their model achieved an average AUC score of 0.92 on hold out sets in a 5 fold cross validation experiment. It did not perform as well as the G4Hunter method (Bedrat et al. 2016), however, when tested on the rG4-seq dataset developed by Kwok et al. 2016, suggesting that some positional information is lost when sequences are converted to trinucleotide content.

Finally, the more recent efforts to produce high-throughput methods for identifying genomic G4s, such as G4-seq developed by Chambers et al. 2015, have created much better in depth datasets for training machine learning models. Sahakyan et al. used the G4-seq dataset in their model. This was a extreme gradient boosted machine model developed using `xgboost`, which regressed the percentage mismatch score of sequences from the G4-seq dataset which conform to the Quadparser method (Sahakyan et al 2017). The authors extracted features from Quadparser conforming PG4s similar to those employed by Stegle et al., including tetrad number, loop length, and mono-, di- and triunucleotide contents of the PG4, and flanking regions. This model was very successful at identifying Quadparser conforming motifs which did or did not actually form G4s, achieving a root mean squared error score of 8.14 (units used were mismatch score in G4-seq experiment, in percentage format) (Sahakyan et al. 2017).

\newpage

## Biological Roles of G-Quadruplexes

### Genome Stability & DNA Replication

The distribution of G4s in genomes has been predicted from sequence for a wide variety of organisms, and has been experimentally determined by the techniques mentioned above for the human genome. There is conclusive evidence that G4s are not uniformly distributed throughout genomes, but tend to be clustered at functional locations. By far the strongest enrichment of G4s is seen at telomeres. G4s are also found more than would be expected in origins of replication, gene promoters, and inside gene bodies, particularly the 5' and 3' untranslated regions (UTRs). In these locations, it has been demonstrated that G4 formation has effects on the processes of DNA replication, genome stability, transcription, and translation.

Telomeres are the protein-DNA structures found at the ends of linear eukaryotic chromosomes. They consist of thousands of tandem repeats of a G-rich sequence (Moysis et al. 1998), with a single stranded overhang of around 100-200bp (Makarov et al. 1997). Due to functional limitations in templated DNA replication, the very ends of these cannot be duplicated during cell division. This means that without intervention, the chromosome will gradually shorten with each division. Telomeres therefore serve as protective caps that prevent the loss of important coding DNA from the genome. In humans, the telomeric repeat is (TTAGGG)n (Moysis et al. 1998). This has been identified through various methods as a G4 forming sequence.

Telomeres are coated in architectural proteins, called telomere end binding proteins (TEBPs), which protect the DNA from recognition by DNA damage response pathways. Giraldo & Rhodes showed that a yeast TEBP, RAP1, induces formation of G4 structures in telomeres *in vitro* (Giraldo & Rhodes, 1994). More recently, immunofluorescence experiments using the BG4 antibody showed that G4 foci overlap with fluorescent foci produced by Fluorescent *In Situ* Hybridisation (FISH) of the telomeric repeat in human HEK 293T cells (Moye et al. 2015). These results suggest that telomeric sequences do form G4s *in vivo*, which may be bound and stabilised by TEBPs. Furthermore, Moye et al. identified that telomerase, the template-independent DNA polymerase which synthesises new telomeric repeats, is able to bind to and partially unwind parallel G4s, but not anti-parallel G4s (Moye et al. 2015). There is therefore the suggestion of a G4 regulated mechanism for telomere maintenance.

G4s seem to also play roles in other aspects of DNA replication. It is well documented that G4s are capable of stalling polymerases *in vitro*. There is also growing evidence that without the assistance of G4 unwinding helicases, G4s might cause polymerase stalling *in vivo*, also. Work by Rodriguez et al. demonstrated that treatment of human cancer cells with pyridostatin causes an increase in the DNA damage marker γH2AX, suggesting an increase in double strand breaks (DSB). This damage was ameliorated by treatment with a DNA replication inhibitor, suggesting the damage was caused during replication. This is likely to be the result of replication fork collapse at G4 DNA blockages. The DNA helicase FANCJ, which is a tumour-suppressor gene often mutated in breast and ovarian cancers, is involved in DSB repair and has been shown to preferentially bind and unwind G4s (Wu & Spies, 2016). Mutation of the FANCJ ortholog DOG1 in *C. elegans* results in genome instability, and accumulation of deletions upstream of G4s (Kruisselbrink et al. 2008).

The Bloom gene (BLM) encodes a DNA helicase which, when mutated in humans, causes Bloom's Disease (German, 1993). This is characterised by a reduction in genome stability resulting from a large increase in sister chromatid exchange events (SCE). SCEs are caused by double strand breaks, which are repaired via homologous recombination (Wu, 2007). BLM has been shown to prevent SCEs by unwinding Holliday Junctions (a common form of branched duplex), and restarting collapsed replication forks (Davies et al. 2007). Work by Sun et al. first showed that BLM is also capable of binding and resolving G4s *in vitro* (Sun et al. 1998). Recently, Wietmarschen et al. used a single cell sequencing technique to map the locations of SCEs in cells lacking function BLM, and found that many SCEs were at sites predicted to form G4s (Wietmarscen et al. 2018). This suggests that direct action of helicases is required to prevent genome instability at G4 loci.

In humans, DNA replication occurs from tens to hundreds of thousands of origins of replication which are found at regular distances of around 10-100kb apart (Huberman and Riggs 1968, Besnard et al. 2012). Using genome-wide mapping of replication origins by short nascent strand sequencing, Besnard et al. identified that the majority of human origins were in GC rich regions of DNA, and that 67% overlapped with motifs conforming to the Quadparser pattern (Besnard et al. 2012). 91% of origins were associated with G4s with loop length of up to 15bp. Furthermore, they found an association between the number of G4 motifs, and the strength of usage of origins, suggesting that G4s might be a recruiting factor for replication machinery.

\newpage

![**Role of G4s in DNA replication** Illustration showing the possible mechanisms of G4 involvement in DNA replication. Besnard et al. identified that a majority of human replication origins contain a PG4, suggesting that G4s may recruit replication machinery (Besnard et al 2012). In the absence of G4 unwinding helicase FANCJ, Kruisselbrink et al. identified deletions downstream of PG4 loci, suggesting G4s cause replication fork collapse. \label{replic}](figures/replication.svg)

\newpage

### Transcription

Transcription is the process by which DNA is copied into messenger RNA (mRNA) or non-coding RNA (ncRNA), by a DNA-templated RNA polymerase. In eukaryotic systems, all mRNA is transcribed by RNA Polymerase II (Pol II). Initiation of transcription is often catalysed by general or specific transcription factors which bind to the promoter region, upstream of the transcriptional start site (TSS). Human promoter sequences are enriched for PG4 motifs conforming to the Quadparser motif (Huppert and Balasubramanian, 2007, Eddy & Maizels 2006). Tumour suppressor genes have fewer promoter PG4s than might be expected by chance, whilst proto-oncogenes contain more than might be expected (Eddy & Maizels 2006). These also overlap with regions of open chromatin, detected by methods such as DNase Hypersensitivity sequencing (DNase-seq) or Assay for Transposable-Accessible Chromatin by Sequencing (ATAC-seq), suggesting they are often actively transcribed (Huppert and Balasubramanian, 2007). Hänsel-Hertsch et al. used the BG4 antibody to perform ChIP-seq of G4 structures in conjunction with ATAC-seq and RNA-seq, in normal human keratinocytes and an immortalised cell line. They found that BG4 peaks were indeed associated with open promoters, and were found upstream of expressed genes (Hänsel-Hertsch et al. 2016). Interestingly, many more BG4 peaks were identified in the immortalised cells than in normal cells, despite having similar levels of open chromatin. Furthermore, genes which only had a promoter BG4 peak in the immortalised cells tended to be more highly expressed in those cells compared to the normal cells, suggesting that promoter G4s may increase gene expression. This increase in expression may be the result of recruitment of positive transcription factors.

Perhaps the most well studied promoter G4 is the Nuclease Hypersensitive Element III (NHEIII) found in the promoter of the c-MYC oncogene. The NHEIII contains a number of G-rich tracts which have been shown to form G4s *in vitro* by a variety of methods (Simonsson et al. 1998, Siddiqui-Jain et al. 2002, Seenisamy et al. 2004, Ambrus et al. 2005). Formation of a G4 by this region has a strongly repressive effect on gene expression. Siddiqui-Jain et al. found that treatment of cells with the G4 stabilising agent TMPyP4 led to repression of c-MYC, whilst G4 abolishing mutations in a c-MYC promoter luciferase assay caused a three-fold increase in expression (Siddiqui-Jain et al. 2004). Furthermore, Grand et al. identified that ~30% of human colorectal tumours have G->A transitions which abolish G4 forming potential *in vitro* (Grand et al. 2004). This was associated with increased expression, indicating that promoter G4s are key in regulation (and mis-regulation) of c-MYC. Pull-down of NHEIII binding factors by González et al. identified Nucleolin as a potential G4 interacting partner (González et al. 2009). Nucleolin is a multi-functional protein implicated in ribosome synthesis, transcription and chromatin remodelling. González et al. went on to show that Nucleolin binds to the c-MYC promoter *in vivo*, and that nucleolin overexpression results in downregulation of c-MYC gene expression.

G4s which form within the gene body may have differing effects on transcription, depending on the strand they occur in. Analysis of human gene expression data has suggested that genes containing coding strand G4s downstream of transcriptional start sites tend to have higher expression at the mRNA level than those that do not (Du et al. 2008), even when other factors such as gene function are controlled for. It has been speculated that G4 formation competes with double stranded DNA, creating single stranded "bubbles" which promote Pol II binding and transcription (Rhodes & Lipps 2015). Since the coding strand is not-directly used by Pol II, coding strand G4s will not cause polymerase stalling. G4s which form in the template strand, however, may form blockages which could slow or pause the progression of Pol II.

Transcription progresses by using the template strand as an antisense copy to replicate the coding strand sequence in mRNA. G4s which form in the template strand of gene body DNA will therefore need to be resolved before Pol II can move through them. Due to the relative stabilities of dsDNA and G4s, G4 formation may only occur in the gene body after a pioneering round of transcription, during which the DNA is in single stranded form. Rodriguez et al. showed that some of the DNA damage caused by treating cells with pyridostatin was transcription-dependent, and could be ameliorated by treating cells additionally with an inhibitor of transcription (Rodriguez et al. 2012).

\newpage


![**Possible Mechanisms for DNA and RNA G4 function in transcription and gene expression** Adapted from Figure 4. Rhodes and Lipps 2015. **a)** Possible mechanism for function of G4s located in the coding strand. Since G4s form from single stranded DNA, G4s in the coding strand may promote melting of double stranded DNA, increasing transcription levels. G4s which form in the coding strand of the exonic DNA of a gene will also be present in the mRNA produced from that locus, and G4s which form in the coding strand of introns will be present in pre-mRNA. These RNA G4s might also influence gene expression through alteration of pre-mRNA splicing, mRNA stability, or translation. **b)** Possible mechanism for the function of G4s located in the template strand. The template strand of genes is scanned by RNA Polymerase II during transcription, and G4s which form in ahead of the transcription complex may cause slowing or stalling if they cannot be correctly resolved. RNA Polymerase II translocation speed is linked to a number of co-transcriptional processes, including splicing. \label{mech}](figures/transcription.svg)

\newpage

Methods for estimating Pol II elongation rates across genes, such as GRO-seq or BruDRB-seq, have associated changes in speed with various features such as specific histone modifications, exon density, and sequence features. Veloso et al. correlated elongation rates from BruDRB-seq data with GC content of genes, and found that genes with higher GC content tended to have slower elongation (Veloso et al. 2014). They hypothesised that this could be due to the greater stability of GC-rich duplexes, which have extra hydrogen bonds. It is also possible, however, that this effect could be partially due to greater numbers of G4s in GC-rich genes.

In human cells, profiling of Pol II occupancy by ChIP-seq has demonstrated that there is a large peak of paused polymerase in the first 30-60bp downstream of the TSS (Jonkers & Lis 2015). This pausing is an important checkpoint ensuring Pol II is correctly modified before elongation begins. Genes which require large and rapid increases in expression in response to environmental stresses, such as heat shock proteins, also have large amounts of paused Pol II which can be activated quickly. During initiation of transcription, Pol II is recruited to the TSS by specific or general transcription factors, and transcribes for a short distance before becoming paused. Formation of paused Pol II, referred to as the Pre-Initiation Complex (PIC), is stabilised by the Negative elongation factor (NELF) and DRB-sensitivity-inducing factor (DSIF), as well as by phosphorylation of the carboxy-terminal domain (CTD) of Pol II at Serine 5. Productive elongation can then be restarted by the action of the positive transcription elongation factor-b (P-TEFb) complex, which phosphorylates NELF and DSIF, causing the former to be released from the PIC, and the latter to switch to becoming a positive elongation factor. P-TEFb also phosphorylates the CTD at Serine 2, which is considered a hallmark modification of active Pol II. How Pol II pausing is regulated is still not clear (Jonkers & Lis 2015, Liu et al. 2015). One hypothesis is that the sequence content of promoters and promoter proximal regions may be important for regulating pausing. A number of promoter motifs, such as the GAGA motif or the downstream promoter element (DPE) have been associated with promoters with high levels of stalling (Hendrix et al. 2008).

It is well established that the promoter proximal regions of genes in many organisms have, on average, higher GC content than the rest of the gene body (Veloso et al. 2014). Eddy et al. identified that the first 200bp downstream of the TSS tends to be more GC-rich in genes with high levels of proximal pausing than in genes which do not exhibit pausing (Eddy et al. 2011). The G4-forming potential of these genes also tended to be greater on the coding strand, meaning that G4 structures might also form in the nascent mRNA. Eddy et al. hypothesised that these 5' mRNA G4s might signal back to the polymerase to produce pausing. A similar mechanism involving an RNA hairpin has been implicated in pausing of *E coli* RNA Polymerase (Toulokhonov et al. 2007).

The enrichment of G4s in promoters and promoter proximal regions suggests that proteins involved in transcriptional complexes may bind specifically to these structures. The general transcription initiation factor complex, TFIIH, contains 11 subunits, and is required for both transcriptional processes and DNA repair through the Nucleotide Excision Repair (NER) pathway (Compe & Egly, 2012). The DNA helicases XPB and XPD are essential components of TFIIH, which catalyse the denaturation of DNA in promoters or around lesions (Coin et al. 2007). Through *in vitro* binding assays, Gray et al. identified XPB and XPD as G4 interacting proteins, which bind G4s preferentially over dsDNA (Gray et al. 2014). XPD was also found to unwind G4s *in vitro*. Gray et al. went on to perform ChIP-seq of XPD and XPB, and showed that it was enriched at TSS loci containing G4s. Approximately 40% of XPD/XPB peaks contained PG4s conforming to the Quadparser pattern (with loop lengths >= 12bp). Furthermore, Hänsel-Hertsch et al. also reported a strong overlap between these XPD/XPB peaks and G4 loci observed by BG4 ChIP-seq (Hänsel-Hertsch et al. 2016). This suggests that TFIIH may be recruited to G4 containing promoters to initiate transcription.

\newpage

### mRNA Processing

Nascent pre-mRNA which is newly transcribed by Pol II must undergo 5' capping, splicing, RNA modification, poly-adenylation and quality control before it can mature into mRNA which is exported to the cytoplasm. Many of these processes occur co-transcriptionally and are tightly co-ordinated to prevent mistakes. Multiple independent studies in different organisms estimate that between 75%-85% of splicing is conducted in a cotranscriptional manner (Armour et al. 2011, Khodor et al. 2011, Girard et al. 2012, Tilgner et al. 2012, Windhager et al. 2012). Oesterreich et al. found that in yeast, 10% of intron splicing is complete when Pol II is only 26bp downstream of the intron acceptor site, and 50% complete when Pol II is 45bp downstream. (Oesterreich et al. 2016). By modifying Pol II to increase its speed 2.3x, they also showed that splicing could become rate limiting when elongation rate is greater. Furthermore, modifications to splice site sequences in a reporter reduced the rate of splicing, presumably by reducing the strength of recognition by snRNAs, and thereby the rate of spliceosome assembly (Oesterreich et al. 2016). This indicates that interplay of splice site strength and Pol II elongation speed determine the relative efficiency of splicing.

It has been estimated that greater than 90% of human genes undergo some form of alternative splicing (Wang et al. 2008). During this process, different donor and acceptor sites compete to be utilised. The most common form of alternative splicing is alternate donor or acceptor usage, where the other site used is constitutive. Other forms include intron retention, where splice sites are simply not used at all, or exon skipping, where constitutive donor and acceptor sites are paired such that an intervening exon is removed from the mature mRNA. Regulation of alternative splicing can occur via protein splicing factors, as well as through changes in Pol II elongation speed. Through this mechanism, alternative splicing may be linked to changes in Pol II speed as it transcribes through template stranded G4s or other DNA structures.

\newpage

![**Pol II elongation speed affects co-transcriptional splicing** **a)** Mechanism for effect of Pol II speed on co-transcriptional splicing: when elongation occurs rapidly (first row), the more canonical but distal acceptor site is used, resulting in the exclusion of the alternate exon (shown in green). When Pol II elongates more slowly (second row), there is more time for the weaker proximal site to be utilised, resulting in the inclusion of the alternate exon. **b)** Example mechanism for how G4s could affect splicing. When G4s are not present (top row), the constitutive splice acceptor is used and the green exonic chunk is excluded. When G4s are formed in the template strand of the DNA (second row), these slow down Pol II elongation, allowing the weaker proximal splice junction to be utilised, and including the alternate exon chunk. (\label{speed_splice}](figures/polii_speed_splicing.svg)

\newpage

Aside from their effects on Pol II speed, G-rich sequences with the potential to form G4s have been implicated as important intron motifs for splicing. These PG4s are predicted in the coding strand, meaning that they could form G4s in either the DNA or the nascent pre-mRNA. Analysis of the first exon-intron boundary of human genes by Eddy & Maizels revealed that about 50% of boundaries contain PG4s within the first 100bp of intronic sequence (Eddy & Maizels 2008). They noted that a number of hnRNP family proteins, such as hnRNP A1 and hnRNP H, bind to G-rich motifs in RNA. hnRNP A1 has been called the "swiss army knife of gene expression", due to its ability to bind both chromatin and mRNA, and its putative roles in transcription, mRNA splicing, telomere maintenance, mRNA export, and translation. Interestingly, hnRNP A1 is capable of binding and unwinding DNA G4s, and has been demonstrated to increase expression of the KRAS and c-MYC oncogenes by resolving repressive G4s in their promoters (Chu et al. 2017). Furthermore, there is evidence that hnRNPs F and H are capable of binding to G-rich RNA sequences to regulate splicing (Xiao et al. 2009). Xiao et al. identified that G-rich sequences downstream of donor splice sites with intermediate levels of homology to the snRNA U1 were strongly conserved. They also noted that the expression of genes with these intermediate splice sites and G-runs was sensitive to the knock-down of hnRNP H (Xiao et al. 2009).

A model gene for the study of this G-rich motif dependent splicing is Bcl-X, a regulator of cell death which has two major spliced forms. The dominant isoform is the longer, Bcl-XL, which is anti-apoptotic. A switch in splicing leads to formation of the shorter form, Bcl-XS, which is pro-apoptotic (Min et al. 1996). This switch involves differential donor site usage in the splicing of the second intron. Garneau et al. showed that alternative splicing of Bcl-X is mediated by hnRNP F/H binding to two exonic G-rich regions, one upstream of the Bcl-XL donor site, and another downstream of the Bcl-XS donor site (Garneau et al. 2005). Mutation of these G-runs abolished hnRNP binding and removed the effect of recombinant hnRNP F treatment on Bcl-X splicing. Weldon et al. recently demonstrated that both of the G-rich regions are capable of forming G4s *in vitro*, and that treatment of *in vitro* splicing assays with ellipticine derived G4 binding agents was able to alter the ratios of the spliced forms. The RNA recognition domain of hnRNP F binds to single stranded DNA, suggesting that G4s modulate splicing by preventing hnRNP binding (Dominguez et al. 2010).

\newpage

### mRNA Stability and Translation

G4s which form in mRNA have the potential to be more stable than those formed in DNA, because RNAs tend to form more complex structures and do not have to compete with double stranded forms. Furthermore, structural studies have suggested that the extra hydroxyl groups in ribose compared to deoxy-ribose allow RNA G4s to form more hydrogen bonds within the Quadruplex structure (Collie et al. 2010). This increases the enthalpic favourability of the RNA G4 whilst also reducing the entropic cost of hydrogen bonds with ordered water molecules. Kwok et al. have used rG4-seq to identify mRNA G4s that form *ex vivo* in the presence of potassium and pyridostatin, by their ability to stall RT. They found a total of 3383 G4s in mRNAs, of which 62% were contained in the 3' UTR, 16% in the 5' UTR and 21% were in the CDS (Kwok et al. 2016). Simulation of RNA folding with the RNA G4 region constrained yielded very different structures on average to folding without constraints, suggesting that G4 formation might act as a molecular switch, changing the overall structure of the mRNA (Kwok et al, 2016). RNA G4s have been implicated in a number of regulatory roles, including translational regulation in the 5' UTR, and mRNA stability in the 3' UTR.

Translational initiation in eukaryotes begins with the binding of a 43S preinitiation complex (PIC) to the 5' cap of the linear mRNA (Hinnebusch 2014). The PIC is pre-loaded with a Methionine aminoacyl-tRNA, and scans along the 5' UTR of the mRNA until the first methionine codon AUG is identified. This identification catalyses the recruitment of the large 60S subunit of the ribosome, to produce a translationally active complex (Hinnebusch 2014). The 5' UTR of the mRNA often contains structures such as hairpins which must be resolved by RNA helicases to allow the passage of the PIC. G4s which form in the 5' UTR can have similar consequences, impeding the scanning of the PIC, or, if close enough to the m7G cap, preventing the loading of the complex onto the mRNA (Bugaut & Balasubramanian 2012).

A number of oncogenes, such as NRAS, BCL-2 and VEGF have been identified as containing 5' UTR G4s (Kumari et al. 2007, Shahid et al. 2010, Morris et al. 2010). When folded, these often to have repressive effects on translation. The KRAS 5' UTR, for example, causes reduced expression when attached to a luciferase reporter, compared to the same 5' UTR with mutations that abolish G4 folding (Kumari et al. 2007). Wolfe et al. showed that treatment of cells with silvestrol, an inhibitor of the RNA helicase EIF4A, caused ribosome stalling on the 5' UTRs of genes containing PG4s (Wolfe et al. 2014). EIF4A has previously been identified as a required helicase for the translation of transcripts with long and structurally complex 5' UTRs (Parsyan et al. 2011). This evidence suggests that 5' UTR G4s can act as negative regulations of translation initiation.

Interestingly, study of the VEGF 5' UTR has identified a G4 which acts as a positive regulator of translation (Morris et al.  2010). VEGF has an extremely long 5' UTR of greater than 1000 bases, which contains two distinct internal ribosome entry sites (IRES). IRES are sequences in the 5' UTR which can control cap independent binding and initiation of ribosome scanning. They are often found in very long and GC-rich 5'UTRs which are more highly structured, and maintain their translation during stresses such as hypoxyia (Jackson 2013). In VEGF, Morris et al. showed that a G4 structure located at IRES-A was essential for internal ribosome initiation (Morris et al. 2010). This indicates that 5' UTR G4s can have complex roles in translational regulation.

G4s in 3' UTRs have the potential to act as *cis* regulatory elements controlling translation efficiency or mRNA stability. One hypothesised method of action is through competition between G4 structures and unstructured UTRs which can bind microRNAs (miRNAs). miRNAs are short (20-25bp) regulatory RNAs which act through complementarity to their target mRNA. Hybridisation of miRNAs to their targets results in changes in mRNA degradation through P-body localisation, or translational repression# (Ambros 2004). The local structure of 3' UTRs has been identified as a key modulator of their binding efficiency (Long et al. 2007). Rouleau et al. performed a characterisation of the overlap between predicted miRNA binding sites and PG4 sequences in human 3' UTRs (Rouleau et al. 2017). The authors found that 54% of PG4s in 3' UTRs overlapped with at least one predicted miRNA target site, though the enrichment or statistical significance of this overlap were not calculated. A candidate mRNA, FADS2, which is regulated by mir331-3p, was tested for G4 formation. Rouleau et al. showed that a 3' UTR sequence which binds mir331-3p also forms a G4, and that G4 formation prevents miRNA binding *in vitro*.

![**G4 function in translation and mRNA stability**](figures/mRNA_stability_translation.svg)

\newpage

### Chromatin Remodelling

Chromatin conformational dynamics are important regulatory mechanisms for gene expression. The major protein component of chromatin is made up of histones, which assemble into the nucleosomes around which DNA is wrapped. Chromatin remodelling is the process by which modifications are made to histones. These modifications include switching of histone isoforms, covalent modifications, and histone sliding. The latter involves the use of energy from ATP hydrolysis to translocate histones along the DNA polymer without ever removing them (Langst & Manelyte 2015).

One class of remodellers is SWI/SNF complexes, which are able to slide or eject nucleosomes. ATRX is an X-linked SWI/SNF family member, which is mutated in ATR-X syndrome, an intellectual disability disorder. ATRX has been shown to locate at heterochromatic and pericentromeric regions, as well as at telomeres (McDowell et al. 1999), regions which are enriched in G-rich PG4 structures. During ATR-X syndrome, global DNA methylation patterns at these regions are disregulated (Gibbons et al. 2000). Downregulation of ATRX during the S-phase of the cell cycle, when DNA replication occurs, results in an increase in the DNA damage response marker gamma-H2AX at telomeric regions (Wong et al. 2010). Heaphy et al. showed that human pancreatic cancers which had lost the expression of ATRX or its partner protein DAXX had elongated telomeres that were characteristic of the Alternative Lengthening of Telomeres pathway. This pathway is telomerase independent, suggesting that ATRX is important for normal telomere repair (Heaphy et al. 2011).

ATRX mutation also effects gene expression. Gibbons et al. identified gene expression changes in the alpha globin gene cluster in response to ATRX mutation (Gibbons et al. 1991). More recently, Law et al. performed ChIP-seq using an ATRX specific antibody to identify other gene targets. They found that ATRX binds specifically to G-rich tandem repeats which form G4s *in vitro*. Furthermore, the strength of the effect of ATRX mutation on gene expression was proportional to the number of tandem G-rich repeats. This suggests that the effect is increased by the number or stability of G4s (Law et al. 2010).

G4 formation is highly dependent upon chromatin status, and G4s isolated by BG4 ChIPseq have been shown to form in regions of open chromatin (Hänsel-Hertsch et al. 2016). Hänsel-Hertsch et al. showed that treatment of cells with the histone deacetylase inhibitors, which cause a relaxation of the chromatin, also increased the number of G4 loci identified by ChIPseq (Hänsel-Hertsch et al. 2016).

## Role of G-Quadruplexes *in planta*

### G-Quadruplex Distribution

Whilst the majority of G4 studies have been performed in mammalian systems, particularly *H. sapiens*, there is a growing body of evidence for the function of G4s in plant species including *Arabidopsis thaliana*, *Zea mays*, and *Oryza sativa*. 

### Translational Regulation

### Plant Development

### Stress Response

## Summary

\newpage