\section*{Abstract} 
\addcontentsline{toc}{chapter}{Abstract}

G-Quadruplexes (G4s) are four stranded DNA structures which form in regions with high GC content and high GC skew. Because of the dependence of G4 structure on specific sequences, it is possible to predict putative G4s (PG4s) throughout genomic sequence. PG4s are non-uniformly distributed in genomes, with higher densities within various genic features, particularly promoters, 5' untranslated regions (UTRs) and coding sequences (CDSs). When they form G4s, these sequences can have a variety of implications for biological processes including replication, transcription, translation and splicing. Here, we introduce a novel method for identifying PG4s, which uses a machine learning approach trained on datasets derived from the high throughput sequencing of G4 structures. We apply this and other techniques, to study the prevalence of PG4s in the genome of *Arabidopsis thaliana*, the model plant. Finally, we study the effect of G4 stabilisation on gene expression in Arabidopsis, using the G-Quadruplex binding agent N-methyl mesoporphyrin (NMM). We identify a family of genes which are strongly downregulated by NMM, and find that they contain large numbers of PG4s in their CDSs.

\subsection*{Note}

The entirety of this thesis, including all the supplementary code listed in the Appendix, can be found on GitHub at [github.com/mparker2/mparker_phd_thesis](https://www.github.com/mparker2/mparker_phd_thesis). It is also indexed on Zenodo at [10.5281/zenodo.1420288](https://www.doi.org/10.5281/zenodo.1420288).

\newpage