# Appendix

This folder contains all the notebooks and code used to generate the figures for the thesis.

## Contents:

### Chapter 2: A Recurrent Neural Network to Predict G Quadruplex Structure [g4seeqer]

1. pass: `00_h`
* pass: `00_h`

### Chapter 3: Global analysis of Predicted G-Quadruplexes in the Arabidopsis thaliana genome [g4_distribution]

1. pass: `00_h`
* pass: `00_h`

### Chapter 4: Global effect of G Quadruplex stabilisation on gene expression [global_nmm]

1. pass: `00_h`
* pass: `00_h`

### Chapter 5: Effect of G-Quadruplexes on expression and splicing of the Extensin gene family [extensins]

1. Differential Expression analysis of NMM vs DMSO RNAseq experiment, plus GOseq Gene Ontology analysis: `01_nmm_dmso_rnaseq_differential_expression.ipynb`
* PG4 enrichment in Gene Ontology groups and comparison with GOseq results from notebook 01: `02_gene_ontology_pg4_enrichment.ipynb`
* Predictions of PG4s in Extensin gene family: `03_extensin_gene_pg4_estimate_table.ipynb`
* Analysis of LRX1 and EXT3 qPCR data: `04_extensin_qpcr.ipynb`
* Analysis of EXT repeat CD spec data: `05_ext_repeat_cd_spectroscopy.ipynb`
* Visualisation of splice splice consensus sequences using weblogos: `06_exitron_splice_junction_logos.ipynb`
* Simulation of expected number of mismapped splice sites using polyester simulated RNAseq data: `07a_splice_simulation.py`; `07b_splicing_simulation.ipynb`
* Dotplot for EXT9 and mappability vs spliced read analyses for Extensins: `08_ext9_dotplot_and_mappability.ipynb`
* Mapping of Sanger sequenced products to TAIR10 genome: `09_sanger_sequencing_splice_variants.ipynb`
* Limma voom analysis of differential junction usage: `10_differential_splicing_analysis.ipynb`
* Analysis of percent spliced reads in Extensins in NMM vs Control: `11_extensin_percent_spliced_nmm_vs_dmso.ipynb`
